clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -248,6 +249,22 @@ class GCEUtil {
     allMIGSInRegion
   }
 
+  static Autoscaler queryZonalAutoscaler(String projectName,
+                                         String zone,
+                                         String serverGroupName,
+                                         GoogleCredentials credentials) {
+    try {
+      return credentials.compute.autoscalers().get(projectName, zone, serverGroupName).execute()
+    } catch (GoogleJsonResponseException e) {
+      // 404 is thrown if the autoscaler does not exist in the given zone. Any other exception needs to be propagated.
+      if (e.getStatusCode() != 404) {
```

Comment:
I've added an additional check to verify that the response was a `404` _and_ the `details` are populated. I also verified that trying to hit a non-existent endpoint leaves the `details` block null.